### PR TITLE
Send stickers from Android Riot

### LINKED ISSUE #2245: Send Stickers
https://github.com/vector-im/riot-meta/issues/151

### PR DESCRIPTION
**Pull request by bmarty**
- Add a "send sticker" option in the send item menu in a room
- Open the StickerPicker widget if found, or propose to install it
- Let the user choose a sticker to send
- Open the Sticker Picker configuration from the sticker picker Screen
- Send the sticker to the room

- **Note**: onCreate() is now final, you have to move its content to initUiAndData(). You also have to implement getLayoutRes(), and getTitleRes() if necessary

Require https://github.com/matrix-org/matrix-android-sdk/pull/269
Fix  #2245

### CODE REVIEW

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 1]
```diff
@@ -2225,6 +2258,53 @@ private void launchFileSelectionIntent() {
...
+            View v = LayoutInflater.from(builder.getContext()).inflate(R.layout.no_sticker_pack_dialog, null);
+
+            builder
+                    .setView(v)
+                    .setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
+
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            // Open integration manager
+                            openIntegrationManagerActivity();
```
Comment (manuroe) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: We should open the integration manager on the sticker packs settings screen, no?

File: vector/src/main/java/im/vector/activity/BugReportActivity.java [hunk 2]
```diff
@@ -73,12 +72,12 @@
     private MenuItem mSendBugReportItem;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-
-        setContentView(R.layout.activity_bug_report);
-        ButterKnife.bind(this);
```
[the remark this answers is not in the document]
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/BugReportActivity.java]: Yes, it's done by the parent Activity RiotAppCompatActivity

File: vector/src/main/java/im/vector/widgets/WidgetsManager.java [hunk 3]
```diff
@@ -64,12 +70,12 @@
...
      * Integration rest url
      */
-    private static final String INTEGRATION_REST_URL = "https://scalar.vector.im";
+    private static final String INTEGRATION_REST_URL = "https://scalar-staging.riot.im/scalar";
 
     /**
      * Integration ui url
      */
-    public static final String INTEGRATION_UI_URL = "https://scalar-staging.riot.im/scalar-web/";
+    public static final String INTEGRATION_UI_URL = "https://scalar-staging.riot.im/scalar-web";
```
[the remark this answers is not in the document]
Comment (bmarty) [reply vector/src/main/java/im/vector/widgets/WidgetsManager.java]: Ok, URLs moved to `donottranslate.xml` resource file (which maybe could be renamed by something like `config.xml` later?

File: vector/src/main/java/im/vector/activity/StickerPickerActivity.kt [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    }
+
+    /**
+     * Compute the URL
+     *
+     * @return the URL
+     */
+    override fun buildInterfaceUrl(scalarToken: String): String? {
+        try {
+            return mWidgetUrl + "?" +
```
[the remark this answers is not in the document]
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/StickerPickerActivity.kt]: Yes, Maybe, but I do not have a `Widget` object here. I prefer to keep it like this for the moment

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: Yes, It should be great, but we do not know the `widgetId`.

**Comment by bmarty** [comment]
Ok, can I'm ready to merge

**Pull request merged**